src/main/messageExtractor/PeripheralAction.java

```diff
@@ -0,0 +1,24 @@
+package messageExtractor;
+
+/**
+ * Wrapper for the action extracted from the message sent to the server. This object is returned by the message
+ * extractor to be sent to the peripheral specified by the ID.
+ * @see MessageExtractorInterface
+ */
+public class PeripheralAction {
```

Comment:
This class could have an `performAction` method, that would also help eliminate the if-else case mentioned below. Just a suggestion though, there are other ways.